File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        lookup = set("aeiou")
        vowels = 0
        output = 0

        for i in range(k):
            char = s[i]
            if (char in lookup):
                vowels += 1

        output = max(vowels, output)

        for i in range(k, len(s)):
            char = s[i]
            if (char in lookup):
                vowels += 1
            if (s[i - k] in lookup):
                vowels -= 1

            output = max(vowels, output)

        return output     

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.js
var maxVowels = function(s, k) {
    let output = 0
    let lookup = new Set("aeiou".split(""))

    function isVowel(char) {
        return lookup.has(char)
    }    

    let count = 0
    for (let i = 0; i < k; i++) {
        let char = s[i]
        if (isVowel(char)) count += 1
    }

    output = count

    for (let i = k; i < s.length; i++) {
        let char = s[i]
        let prev = s[i - k]
        if (isVowel(prev)) count -=1
        if (isVowel(char)) count += 1
        output = Math.max(output, count)
    }
    return output
};